default.nix

```diff
@@ -1,130 +1,18 @@
-  });
-
-  # declares a build environment where C and C++ compilers are delivered by the llvm/clang project
-  # in this version build process should rely only on clang, without access to gcc
-  llvm = nixpkgs.llvmPackages_11;
-  env = llvm.stdenv;
-  llvmVersionString = "${nixpkgs.lib.getVersion env.cc.cc}";
-
-  # allows to skip files listed by .gitignore
-  # otherwise `nix-build` copies everything, including the target directory
-  gitignoreSrc = nixpkgs.fetchFromGitHub {
-    owner = "hercules-ci";
-    repo = "gitignore.nix";
-    rev = "5b9e0ff9d3b551234b4f3eb3983744fa354b17f1";
-    sha256 = "o/BdVjNwcB6jOmzZjOH703BesSkkS5O7ej3xhyO8hAY=";
+  versions = import ./nix/versions.nix { inherit rocksDBVersion; };
+  alephNode = (import ./nix/aleph-node.nix { inherit versions; }).project;
+  workspaceMembers = builtins.mapAttrs (_: crate: crate.build.override { inherit runTests; }) alephNode.workspaceMembers;
+  allWorkspaceMembers = nixpkgs.symlinkJoin {
+      name = "all-workspace-members";
+      paths = builtins.attrValues workspaceMembers;
   };
-  inherit (import gitignoreSrc { inherit (nixpkgs) lib; }) gitignoreSource;
 in
-with nixpkgs; env.mkDerivation rec {
-  name = "aleph-node";
-  src = gitignoreSource ./.;
-
-  buildInputs = [
-    rustToolchain
-    llvm.clang
-    openssl.dev
-    protobuf
-    customRocksdb
-    pkg-config
-    cacert
-    git
-    findutils
-    patchelf
-  ];
-
-  shellHook = ''
-    export RUST_SRC_PATH="${rustToolchain}/lib/rustlib/src/rust/src"
-    export LIBCLANG_PATH="${llvm.libclang.lib}/lib"
-    export PROTOC="${protobuf}/bin/protoc"
-    export CFLAGS=" \
-        ${"-isystem ${llvm.libclang.lib}/lib/clang/${llvmVersionString}/include"} \
-        $CFLAGS
-    "
-    export CXXFLAGS+=" \
-        ${"-isystem ${llvm.libclang.lib}/lib/clang/${llvmVersionString}/include"} \
-        $CXXFLAGS
-    "
-    # From: https://github.com/NixOS/nixpkgs/blob/1fab95f5190d087e66a3502481e34e15d62090aa/pkgs/applications/networking/browsers/firefox/common.nix#L247-L253
-    # Set C flags for Rust's bindgen program. Unlike ordinary C
-    # compilation, bindgen does not invoke $CC directly. Instead it
-    # uses LLVM's libclang. To make sure all necessary flags are
-    # included we need to look in a few places.
-    export BINDGEN_EXTRA_CLANG_ARGS=" \
-        ${"-isystem ${llvm.libclang.lib}/lib/clang/${llvmVersionString}/include"} \
-        $BINDGEN_EXTRA_CLANG_ARGS
-    "
-    export ROCKSDB_LIB_DIR="${customRocksdb}/lib"
-    export ROCKSDB_STATIC=1
-  '';
-
-  buildPhase = ''
-    ${shellHook}
-    export CARGO_HOME="$out/cargo"
-    export CARGO_BUILD_TARGET="x86_64-unknown-linux-gnu"
-
-    cargo build --locked --release -p aleph-node
-  '';
-
-  installPhase = ''
-    mkdir -p $out/bin
-    mv target/x86_64-unknown-linux-gnu/release/aleph-node $out/bin/
-  '';
-
-  fixupPhase = ''
-    find $out -type f -exec patchelf --shrink-rpath '{}' \; -exec strip '{}' \; 2>/dev/null
-  '';
-}
+if allCrates then
```

Comment:
I think this flag's logic would be more apparent when flipped, i.e.
```
if alephNodeOnly then
    workspaceMembers."aleph-node"
else
    allWorkspaceMembers
```

Reply:
actually I would prefer to have something like `crates = [ "aleph-node" ]` as parameter, where for all crates there would be a special value `[ ]`

Reply:
Oh sorry, forgot to reply since I agree with you :joy: